src/platform/packages/private/opentelemetry/kbn-metrics/README.md

```diff
@@ -92,3 +111,24 @@ When shipping the metric through an OTLP exporter, the document stored in ES loo
   }
 }
 ```
+
+### Naming convention
+
+It is highly recommended to adhere to OTel's official naming convention: https://opentelemetry.io/docs/specs/semconv/general/naming/
+
+A few highlights:
+
+1. Double-check the OTel docs to identify if your metric already has an official name.
+2. Use dot-notation to separate the different parts of the metric name. Snake_case is also allowed in some cases.
+3. Typically, [these instrument names](https://opentelemetry.io/docs/specs/semconv/general/naming/#instrument-naming) should cover most of the metrics.
+
+### Which metric type should I use?
+
+The [OTel docs](https://opentelemetry.io/docs/concepts/signals/metrics/) provide a good overview of the different types of metrics.
+
+A quick summary:
+
+1. Use Histogram when you are able to record your metric in push-based mode, and when you're interested in value statistics like _How many requests took less than 1s?_.
+2. The difference between a Counter and an UpDownCounter is that the former is an always growing metric, while the latter can increase or decrease.
+3. Use a Counter (or UpDownCounter) instead of a Gauge if your metric can be summed up (e.g. number of requests or process' memory usage). 
+4. Use a Gauge when it doesn't make sense to sum up the metric (e.g. process' uptime or event loop delay).
```

Comment:
Curious if there's value in highlighting the types:

```suggestion
1. Use `Histogram` when you are able to record your metric in push-based mode, and when you're interested in value statistics like _How many requests took less than 1s?_.
2. The difference between a `Counter` and an `UpDownCounter` is that the former is an always growing metric, while the latter can increase or decrease.
3. Use a `Counter` (or `UpDownCounter`) instead of a `Gauge` if your metric can be summed up (e.g. number of requests or process' memory usage). 
4. Use a `Gauge` when it doesn't make sense to sum up the metric (e.g. process' uptime or event loop delay).
```